base/arrayshow.jl

```diff
@@ -320,14 +320,21 @@ print_array(io::IO, X::AbstractArray) = show_nd(io, X, print_matrix, true)
 # typeinfo aware
 # implements: show(io::IO, ::MIME"text/plain", X::AbstractArray)
 function show(io::IO, ::MIME"text/plain", X::AbstractArray)
+    # -1) handle arrays-in-array (recursive call)
+    if get(io, :compact, false) === true
+        # Using `show` of method with the most abstract type to avoid infinite recursion.
+        invoke(show, Tuple{IO,AbstractArray}, io, X)
+        return
+    end
+
     # 0) show summary before setting :compact
     summary(io, X)
     isempty(X) && return
     print(io, ":")
     show_circular(io, X) && return
 
     # 1) compute new IOContext
-    if !haskey(io, :compact) && length(axes(X, 2)) > 1
```

Comment:
I'm not sure if people are happy with removing `length(axes(X, 2)) > 1`.

(BTW I think this is a shortcoming of the way `:compact` is used.  It's not possible to declare compactness of the width and height separately.  This is why I've been suggesting to use `:displaysize` instead of `:compact` https://github.com/JuliaLang/julia/issues/33260#issuecomment-531991052)

Reply:
This is a separate issue and will probably be controversial, so I wouldn't change this for now.